### PR TITLE
pip install --user flake8

### PR DESCRIPTION
**Pull request by cclauss**
flake8 is a great way to find issues in Python code.  It will help us to find issues and changes so that code will run in both Python 2 and Python 3.

### CODE REVIEW

**Comment by mrutkows** [comment]
@cclauss shouldn't max. line length be according to PEP8; otherwise, it will not considered Pythonic by Python developers?  See https://www.python.org/dev/peps/pep-0008/#maximum-line-length

**Comment by cclauss** [comment]
79 characters is fewer characters than we used to get on punch cards ;-)

Let's start off lenient and then turn the screws later on.  Right now if you click on the `before script` twisty in https://travis-ci.org/openwhisk/openwhisk/builds/208265761 you will see that flake8 currently kicks out **438 errors** and 34 of those are lines longer than 127 characters!  Once we bring those numbers down, we could consider lowering maximum-line-length to the punch card range.

**Comment by mrutkows** [comment]
@cclauss I agree. When being exposed to the serious Python world, I was not pleased with most of PEP8 and especially the 79 char limitation/convention.  Again, please read my concession.

**Comment by rabbah** [comment]
@cclauss is this ready?

**Comment by cclauss** [comment]
Yes, please.

**Comment by rabbah** [comment]
OK - one q: is the change to `core/actionProxy/Dockerfile` necessary in this pr?
(will decide if I should run trigger a Jenkins build)

**Comment by rabbah** [comment]
Also note you don't need to rebase/merge to master when out of date - will take care of that for you ;)

**Comment by cclauss** [comment]
We can pull the Docker mods out for now.

Thanks for the tip on the rebase/merge.

File: tools/travis/setup.sh [hunk 1]
```diff
@@ -14,8 +14,11 @@ docker info
 
 # Python
 sudo apt-get -y install python-pip
+pip install --upgrade pip setuptools
 pip install --user jsonschema
 pip install --user couchdb
+pip install --user backports.ssl_match_hostname
```
Comment (rabbah) [inline tools/travis/setup.sh]: is this related to one of the other prs?

**Comment by rabbah** [comment]
Here's the commit as I will merge it: https://github.com/openwhisk/openwhisk/compare/master...rabbah:pr-1954?expand=1.

File: tools/travis/setup.sh (hunk 1 above)
Comment (cclauss) [inline tools/travis/setup.sh]: https://travis-ci.org/openwhisk/openwhisk/builds/208065811 has the line...

fatal: [172.17.0.1]: FAILED! => {"changed": false, "failed": true, "msg": "Failed to import docker-py - No module named ssl_match_hostname. Try `pip install docker-py`"}

This was probably related to the broken build this morning so `backports.ssl_match_hostname` might no longer be needed.

**Comment by rabbah** [comment]
Merged as https://github.com/openwhisk/openwhisk/commit/2358c0977ff03afee123468cb513dea3706a4d1f.
